### PR TITLE
adding package streamdisc-server

### PR DESCRIPTION
**Pull request by tilwolff**
streamdisc-server is a new addon in the section service/multimedia.

use it to watch a dvd or blu ray on your raspberry, wetek play or any other device without optical drive while the physical disc is in the optical drive of your main openelec pc .

### CODE REVIEW

**Comment by lrusak** [comment]
Looks pretty good :+1: 

You should split the libs into separate packages. Just put them into the lib directory.
libdvdcss
libdvdread

Also, I see you are the owner of the streamdisc repo, so you should create releases on there and have the PKG_VERSION follow that.

File: addons/service/multimedia/streamdisc-server/source/bin/streamdisc-server.start [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+oe_setup_addon service.multimedia.streamdisc-server
+
+ADDON_SETTINGS="$ADDON_HOME/settings.xml"
+
+chmod a+x $ADDON_DIR/bin/*
+
+export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$ADDON_DIR/lib"
+
+exec $ADDON_DIR/bin/streamdisc_server $STREAMDISC_PORT $STREAMDISC_DEVICE
```
Comment (lrusak) [inline addons/service/multimedia/streamdisc-server/source/bin/streamdisc-server.start]: Where does STREAMDISC_PORT and STREAMDISC_DEVICE come from?

File: addons/service/multimedia/streamdisc-server/source/bin/streamdisc-server.start [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#  along with OpenELEC.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+. /etc/profile
+
+oe_setup_addon service.multimedia.streamdisc-server
+
+ADDON_SETTINGS="$ADDON_HOME/settings.xml"
+
+chmod a+x $ADDON_DIR/bin/*
```
Comment (lrusak) [inline addons/service/multimedia/streamdisc-server/source/bin/streamdisc-server.start]: Not needed, see https://github.com/OpenELEC/OpenELEC.tv/blob/master/packages/mediacenter/kodi/profile.d/99-kodi.conf

File: addons/service/multimedia/streamdisc-server/source/bin/streamdisc-server.start [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$ADDON_DIR/lib"
```
Comment (lrusak) [inline addons/service/multimedia/streamdisc-server/source/bin/streamdisc-server.start]: See above comment

**Comment by tilwolff** [comment]
Thanks, will do.

I was not aware it was possible to separate addons into several 
packages. Just realized I could just have used "get_build_dir 
libdvdread" in the "addon()" function and use the default build 
instructions for the packages.

On 11/27/15 21:07, Lukas Rusak wrote:
>
> Looks pretty good :+1:
>
> You should split the libs into separate packages. Just put them into 
> the lib directory.
> libdvdcss
> libdvdread
>
> Also, I see you are the owner of the streamdisc repo, so you should 
> create releases on there and have the PKG_VERSION follow that.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/OpenELEC/unofficial-addons/pull/88#issuecomment-160198857>.
>

-- 
---------------------------
tilman wolff-siemssen
kopischstr. 3
10965 berlin

fon +49 179 900 1074

File: addons/service/multimedia/streamdisc-server/source/bin/streamdisc-server.start (hunk 1 above)
Comment (tilwolff) [inline addons/service/multimedia/streamdisc-server/source/bin/streamdisc-server.start]: STREAMDISC_PORT and STREAMDISC_DEVICE are supposed to come from the 
addon settings (source/settings_default.xml, 
source/resources/settings.xml) and are set by

oe_setup_addon service.multimedia.streamdisc-server. Works on my system...

On 11/27/15 21:13, Lukas Rusak wrote:
>
> In 
> addons/service/multimedia/streamdisc-server/source/bin/streamdisc-server.start 
> <https://github.com/OpenELEC/unofficial-addons/pull/88#discussion_r46068766>:
>
> > +#
> > +#  You should have received a copy of the GNU General Public License
> > +#  along with OpenELEC.  If not, see <http://www.gnu.org/licenses/>.
> > +################################################################################
> > +
> > +. /etc/profile
> > +
> > +oe_setup_addon service.multimedia.streamdisc-server
> > +
> > +ADDON_SETTINGS="$ADDON_HOME/settings.xml"
> > +
> > +chmod a+x $ADDON_DIR/bin/*
> > +
> > +export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$ADDON_DIR/lib"
> > +
> > +exec $ADDON_DIR/bin/streamdisc_server $STREAMDISC_PORT $STREAMDISC_DEVICE
>
> Where does STREAMDISC_PORT and STREAMDISC_DEVICE come from?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/OpenELEC/unofficial-addons/pull/88/files#r46068766>.
>

-- 
---------------------------
tilman wolff-siemssen
kopischstr. 3
10965 berlin

fon +49 179 900 1074

**Comment by tilwolff** [comment]
done. what do I put into PKG_MAINTAINER??

**Comment by chewitt** [comment]
@tilwolff - It should be a method for contacting you, e.g. email address, website with contact form, or an OE or Kodi forum nick. Use mild obfuscation if you want/need to, e.g. user (at ) domain (dot) tld

**Comment by chewitt** [comment]
looks good to me - @lrusak any final objections?

**Comment by lrusak** [comment]
@tilwolff can you squash the commits into one commit per package please? Then I can merge.

File: addons/service/multimedia/streamdisc/changelog.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+0.1
```
Comment (lrusak) [inline addons/service/multimedia/streamdisc/changelog.txt]: should be 6.0.1

**Comment by tilwolff** [comment]
all commits squashed into one...

File: addons/service/multimedia/streamdisc/package.mk [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+PKG_DEPENDS_TARGET="toolchain libbluray libdvdread"
+PKG_PRIORITY="optional"
+PKG_SECTION="service/multimedia"
+PKG_SHORTDESC="streamdisc (Version: $PKG_VERSION): stream unencrypted DVD and BluRay content to clients that do not have an optical drive."
+PKG_LONGDESC="streamdisc (Version: $PKG_VERSION): a tiny http streaming server delivering unencrypted DVD and BluRay content to clients that do not have an optical drive. On the client side, just add http://your.server.ip.address:port as a new generic kodi source."
+PKG_IS_ADDON="yes"
+PKG_ADDON_TYPE="xbmc.service"
+PKG_ADDON_PROVIDES=""
+PKG_AUTORECONF="no"
+PKG_ADDON_REPOVERSION="6.0"
```
Comment (lrusak) [inline addons/service/multimedia/streamdisc/package.mk]: should be 7.0 now

File: addons/service/multimedia/streamdisc/package.mk [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with OpenELEC.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+PKG_NAME="streamdisc"
+PKG_VERSION="0.1"
+PKG_REV="1"
```
Comment (lrusak) [inline addons/service/multimedia/streamdisc/package.mk]: should be PKG_REV="0"

File: addons/service/multimedia/streamdisc/changelog.txt (hunk 4 above)
Comment (lrusak) [inline addons/service/multimedia/streamdisc/changelog.txt]: should be 7.0.0 now

**Comment by lrusak** [comment]
just a couple final fixups with the versioning and it should be good to go.

File: addons/service/multimedia/streamdisc/changelog.txt (hunk 4 above)
Comment (tilwolff) [inline addons/service/multimedia/streamdisc/changelog.txt]: Does that mean The addon wont be available for OE6 Users ?

Von meinem Mobiltelefon gesendet

Comment (lrusak) [inline addons/service/multimedia/streamdisc/changelog.txt]: I'll make a backport once it's in master

**Comment by tilwolff** [comment]
fixed up versioning and squashed again

**Comment by lrusak** [comment]
Thanks for your contribution :)

**Pull request merged**